src/plugins/dashboard/server/saved_objects/dashboard_migrations.ts

```diff
@@ -216,16 +217,12 @@ export interface DashboardSavedObjectTypeMigrationsDeps {
 export const createDashboardSavedObjectTypeMigrations = (
   deps: DashboardSavedObjectTypeMigrationsDeps
 ): SavedObjectMigrationMap => {
-  const embeddableMigrations = Object.fromEntries(
-    deps.embeddable
-      .getMigrationVersions()
-      .filter((version) => semver.gt(version, '7.12.0'))
-      .map((version): [string, SavedObjectMigrationFn] => {
-        return [version, migrateByValuePanels(deps, version)];
-      })
-  );
+  const embeddableMigrations = (mapValues(
+    deps.embeddable.getAllMigrations(),
+    migrateByValuePanels
+  ) as any) as MigrateFunctionsObject;
```

Comment:
`as any` is usually a code smell.  I know mapping objects can be hairy at times, but I'd prefer this be a `const` that's typed, rather than an inline `as any`... and it deserves a comment.

Reply:
fixed